### PR TITLE
OSDOCS-495: Added docs for configuring additionalCORSAllowedOrigins

### PR DESCRIPTION
**Pull request by bergerhoffer**
Preview: http://file.rdu.redhat.com/~ahoffer/2019/OSDOCS-495/authentication/allowing-javascript-access-api-server.html

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@stlaz Can you take a look at this first draft for configuring additionalCORSAllowedOrigins?

http://file.rdu.redhat.com/~ahoffer/2019/OSDOCS-495/authentication/allowing-javascript-access-api-server.html

File: modules/auth-allowing-javascript-access-api-server.adoc [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Module included in the following assemblies:
+//
+// * authentication/configuring-corsallowedorigins.adoc
+
+[id="auth-allowing-javascript-access-api-server_{context}"]
+= Allowing JavaScript-based access to the API server from additional hosts
+
+By default, localhost and 127.0.0.1 are whitelisted to access the API server and
```
Comment (stlaz) [inline modules/auth-allowing-javascript-access-api-server.adoc]: This seems a bit too technical, and it's also not all correct, 127.0.0.1 and localhost are specific for the kube-apiserver configuration, but the oauth-server does not have default CORS set.

You can say that the default configuration allows only the OpenShift console to send requests to the API server.

File: modules/auth-allowing-javascript-access-api-server.adoc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  generation: 1
+  name: cluster
+  resourceVersion: "907"
+  selfLink: /apis/config.openshift.io/v1/apiservers/cluster
+  uid: 4b45a8dd-a402-11e9-91ec-0219944e0696
+spec:
+  additionalCORSAllowedOrigins:
+  - (?i)//my\.subdomain\.domain\.com(:|\z) <1>
+----
+<1> The host name is specified as a Golang regular expression that matches
```
Comment (stlaz) [inline modules/auth-allowing-javascript-access-api-server.adoc]: Should we include the Golang regexp documentation link? https://github.com/google/re2/wiki/Syntax

File: modules/auth-allowing-javascript-access-api-server.adoc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Edit the API servers resource.
++
+----
+$ oc edit apiservers.config.openshift.io
```
Comment (stlaz) [inline modules/auth-allowing-javascript-access-api-server.adoc]: `oc edit apiserver.config.openshift.io cluster`

File: modules/auth-allowing-javascript-access-api-server.adoc (hunk 2 above)
Comment (bergerhoffer) [reply modules/auth-allowing-javascript-access-api-server.adoc]: Unfortunately we've been asked not to link to GitHub repos in the product docs. Unless @sferich888 wants to make an exception here.

**Comment by bergerhoffer** [comment]
@stlaz I've made the adjustments here, look okay now?

http://file.rdu.redhat.com/~ahoffer/2019/OSDOCS-495/authentication/allowing-javascript-access-api-server.html

File: modules/auth-allowing-javascript-access-api-server.adoc (hunk 2 above)
Comment (sferich888) [reply modules/auth-allowing-javascript-access-api-server.adoc]: This link here is OK, as it's referencing an uncommon regex syntax (would be good to get a link version locked to what we build with). Or better if we could pull over the necessary docs for the syntax parser.

Comment (bergerhoffer) [reply modules/auth-allowing-javascript-access-api-server.adoc]: Great thanks @sferich888!

FYI @stlaz I've added this link now. Let me know if there's a particular version of this wiki content that you think we should link to, that matches the version that OpenShift currently supports.

Comment (stlaz) [reply modules/auth-allowing-javascript-access-api-server.adoc]: I think we should be ok with the page here. The set of regular expressions of a language does not change that often (if ever), I think we should be safe with whatever appears on the upstream wiki.

**Comment by stlaz** [comment]
Looks good to me

**Review by stlaz — APPROVED** [review]

**Comment by bergerhoffer** [comment]
@openshift/team-documentation For peer review please.

File: modules/auth-allowing-javascript-access-api-server.adoc [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+application using a different host name, you can configure additional host names
+to allow.
+
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Edit the API servers resource.
```
Comment (bmcelvee) [inline modules/auth-allowing-javascript-access-api-server.adoc]: s/resource./resource:

File: modules/auth-allowing-javascript-access-api-server.adoc [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
++
+----
+$ oc edit apiserver.config.openshift.io cluster
+----
++
+. Add the `additionalCORSAllowedOrigins` field under the `spec` section and
+specify one or more additional host names.
```
Comment (bmcelvee) [inline modules/auth-allowing-javascript-access-api-server.adoc]: s/names./names:

**Comment by bmcelvee** [comment]
Just a couple of suggestions. Otherwise, LGTM.

**Comment by bergerhoffer** [comment]
Thanks @bmcelvee!

**Comment by imdmahajankanika** [comment]
Hello @bergerhoffer, this solution is only working for Chrome, in Firefox and even from chrome on cellphone, there is still a CORS issue.

**Comment by bergerhoffer** [comment]
Hi @imdmahajankanika, since this PR is so old, can you please contact [Red Hat Support](https://docs.openshift.com/container-platform/4.13/support/getting-support.html) or open a bug to report the issue that you're seeing?

**Pull request merged**